_posts/2022-05-02-the-na-bug/the-na-bug.Rmd

```diff
@@ -0,0 +1,111 @@
+```
+
+```{r load_date, echo = FALSE}
+library(here)
+library(xts)
+library(dygraphs)
+library(dplyr)
+library(ggplot2)
+data <- read.csv(file.path(here::here(), "_posts/2022-05-02-the-na-bug", "daily_electrolyte_counts.csv"), stringsAsFactors = FALSE) %>%
+  mutate(date = as.Date(date))
+
+daily_counts_pre_fix <- data %>%
+    filter(cohort == "pre-fix")
+
+daily_counts_post_fix <- data %>%
+    filter(cohort == "post-fix")
+```
+
+## Silent deployment
+
+Our team had been working endlessly on developing [CHARTwatch](https://www.chartdatascience.ca/the-projects/an-early-warning-system-for-general-internal-medicine-), an early warning system and in November 2019, we were ready to move to our **silent deployment** phase. This phase involved deploying our system and, as the name suggests it, running it _silently_. More concretely, our entire pipeline was running (from data extraction to data processing to model prediction), but no outputs were going to the end-user. 
+
+Typically, the goal of the silent deployment phase is to uncover unexpected behaviors with the data, system, or model. During model development and evaluation, we had only worked with _historical extracts of the data_. When moving from historical data to _live data_, there's the risk of running into data issues [@cohen_problems_2021].
+
+- The data can be different due to **external factors**. For example, all of our models were trained on data prior to COVID-19, but shortly after the beginning of our silent deployment phase, we began to observe cases of COVID-19 in the hospital.
+
+- The data can be different due to **data entry errors**. For example, a body temperature could incorrectly be entered as 3700 °C instead of 37.00 °C.
+
+- The data can be different due to **selection bias**. For example, during training we excluded patients with really short and really long visits, as they were rare. However, we may encounter these kinds of visits in the live data.
+
+
+## Monitoring labs
+
+We had set up a monitoring dashboard to measure model inputs and model outputs. On close inspection, we made a discovery that was unquestioNAbly odd... no sodium labs had been measured since we had moved to silent testing!
+
+```{r missingna, fig.cap="Daily counts of lab measurements: this includes counts for calcium (CA), chloride (CL), glucose (GLPOC), potassium (K), and sodium (NA)."}
+daily_counts_pre_fix %>%
+    select(-date, -cohort) %>%
+    xts(order.by = daily_counts_pre_fix$date) %>%
+    dygraph(main = "Electrolyte counts")  %>%
+    dyOptions(stackedGraph = TRUE) %>%
+    dyLegend(width = 400)
+```
+
+This made NA sense at all. Sodium is measured in routinely ordered blood tests. It'll usually get ordered alongside other tests (such as calcium, chloride, glucose, and potassium) as part of a [basic metabolic panel](https://www.healthline.com/health/sodium-blood). In Figure \@ref(fig:missingna), we look at the daily counts of labs on units in which CHARTwatch was silently deployed. 
```

Comment:
I like the puns 🧂

Reply:
Maybe unquestioNAbly to unquestio**na**bly for easier reading?